pkg/controllers/core/deployer_controller.go

```diff
@@ -580,17 +607,32 @@ func (r *DeployerReconciler) constructLabelsForDeployer(deployer *corev1alpha1.D
 	return labels
 }
 
+func (r *DeployerReconciler) constructHostForDeployer(deployer *corev1alpha1.Deployer, cm *corev1.ConfigMap) string {
+	domain := defaultDomain
+	if d := cm.Data[defaultDomainKey]; d != "" {
+		domain = d
+	}
+
+	return fmt.Sprintf("%s.%s", deployer.Name, domain)
+}
+
 func (r *DeployerReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	enqueueTrackedResources := func(t apis.Resource) handler.EventHandler {
+	enqueueTrackedResources := func(t runtime.Object) handler.EventHandler {
 		return &handler.EnqueueRequestsFromMapFunc{
 			ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
-				requests := []reconcile.Request{}
-				key := tracker.NewKey(
-					t.GetGroupVersionKind(),
-					types.NamespacedName{Namespace: a.Meta.GetNamespace(), Name: a.Meta.GetName()},
-				)
-				for _, item := range r.Tracker.Lookup(key) {
-					requests = append(requests, reconcile.Request{NamespacedName: item})
+				var requests []reconcile.Request
+
+				versionKinds, _, err := r.Scheme.ObjectKinds(t)
+				if err != nil {
+					r.Log.Error(err, "Unable to detect version kind")
```

Comment:
let's panic here, it should never happen and is indicative of a programming error.